training/security.md

```diff
@@ -0,0 +1,161 @@
+
+We also assume you have a local, insecure 3-node cluster running. If you don't, you can skip the **Tear down existing cluster** section of this lab.
+
+### Overview
+
+1. Tear down existing cluster.
+2. Generate certificate files.
+3. Restart cluster with SSL encryption.
+4. Connect to the cluster using encryption.
+
+### Step 1. Tear Down Cluster
+
+You can do this elegantly by using `cockroach quit` on each node, but we're going to close down everything in one fell swoop:
+
+~~~ shell
+$ pkill -9 cockroach
+~~~
+
+### Step 2. Generate Certificate Files
+
+We're going to generate:
+
+- A certificate authority (CA) to sign all certificates
+- One node certificate per CockroachDB node
+- One client cert per client
+
+#### Certificate Authority
+
+~~~ shell
+$ mkdir certs
+
+$ mkdir my-safe-directory
+~~~
+
+We want to ensure `my-safe-directory`, which will ultimately contain your CA key, is secure, so we should also modify its permissions:
+
+~~~
+chown my-safe-directory
+~~~
+
+Now, create the key pair:
+
+~~~ shell
+cockroach cert create-ca \
+--certs-dir=certs \
+--ca-key=my-safe-directory/ca.key
+~~~
+
+#### Node Certificate
+
+Because we're using a local deployment and all nodes use the same hostname (`localhost`), we only need to generate a single node certificate. Note that this is different than how you would normally deploy certificates in a production environment.
+
+~~~ shell
+$ cockroach cert create-node \
+localhost \
+$(hostname) \
+--certs-dir=certs \
+--ca-key=my-safe-directory/ca.key
+~~~
+
+#### Client Certificates
+
+We'll create client certificates for both the `root` and `itsme` users.
+
+~~~ shell
+$ cockroach cert create-client \
+root \
+--certs-dir=certs \
+--ca-key=my-safe-directory/ca.key
+~~~
+
+~~~
+$ cockroach cert create-client \
+itsme \
+--certs-dir=certs \
+--ca-key=my-safe-directory/ca.key
+~~~
+
+### Step 3. Restart Our Cluster as Secure
+
+With all of our certificates in place, we can now start our cluster securely.
+
+1. Start all 3 nodes:
+    
+    ~~~ shell
+    $ cockroach start \
+    --certs-dir=certs \
+    --host=localhost \
+    --http-host=localhost \
+    --join=localhost:26257, localhost:26258, localhost:26259
+    ~~~
+    
+    ~~~ shell
+    $ cockroach start \
+    --certs-dir=certs \
+    --store=node2 \
+    --host=localhost \
+    --port=26258 \
+    --http-port=8081 \
+    --http-host=localhost \
+    --join=localhost:26257, localhost:26258, localhost:26259
+    ~~~
+    
+    ~~~ shell
+    $ cockroach start \
+    --certs-dir=certs \
+    --store=node3 \
+    --host=localhost \
+    --port=26259 \
+    --http-port=8082 \
+    --http-host=localhost \
+    --join=localhost:26257, localhost:26258, localhost:26259
+    ~~~
+    
+2. Initialize the cluster:
+
+    ~~~
+    $ cockroach init --certs-dir=certs
+    ~~~
+
+### Step 4. Access the Cluster Securely
+
+~~~ shell
+$ cockroach sql \
+--user=itsme
```

Comment:
If this is a new cluster, this user doesn't exist yet...

Reply:
Also, need an `\` at the end of the second line.

Reply:
We're restarting the same cluster, so the user exists from the Identity and Access Management section. Added a note about creating the user if DNE, though.